### PR TITLE
[insights-agent] Exempt polaris check for prometheus sever

### PR DESCRIPTION
**Pull request by makoscafee**
**Why This PR?**
_a short description of why this PR is needed_

Fixes #

**Changes**
Changes proposed in this pull request:

*
*

**Checklist:**

* [ ] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [ ] Any new values are backwards compatible and/or have sensible default.
* [ ] Any new values have been added to the README for the Chart, or helm-docs has been run for the charts that support it.

### CODE REVIEW

File: stable/insights-agent/values.yaml [hunk 1]
```diff
@@ -307,6 +320,28 @@ prometheus:
...
+      runAsNonRoot: true
+      runAsGroup: 65534
+      fsGroup: 65534
+      allowPrivilegeEscalation: false
+      capabilities:
+        drop:
+          - ALL
+    podAnnotations:
+      polaris.fairwinds.com/insecureCapabilities-exempt: "true"
+      polaris.fairwinds.com/privilegeEscalationAllowed-exempt: "true"
```
Comment (rbren) [inline stable/insights-agent/values.yaml:325]: It looks like we can drop these based on the security context you're setting

**Review by rbren — COMMENTED** [review]
Thanks Barnie!

Were you able to verify that the report still runs correctly?

File: stable/insights-agent/values.yaml (hunk 1 above)
Comment (makoscafee) [reply stable/insights-agent/values.yaml:325]: Apparently those are `podSecurityContext` and not `containerSecurityContext` so 
```
allowPrivilegeEscalation: false
        capabilities:
          drop:
            - ALL
```
does not work for podSecurityContext.

Comment (makoscafee) [reply stable/insights-agent/values.yaml:325]: removed security context for `prometheus-server`

**Comment by makoscafee** [comment]
and verified everthing is working -> https://staging.insights.fairwinds.com/v0/organizations/acme-co/clusters/umoja/reports/resource-metrics/detail?timestamp=2021-10-27T16:31:01.666

**Comment by rbren** [comment]
@makoscafee can you install polaris and kubesec in that staging environment to see if the action items got fixed?

File: stable/insights-agent/values.yaml [hunk 2]
```diff
@@ -296,6 +296,19 @@ resourcemetrics:
 prometheus:
   kubeStateMetrics:
     enabled: true
+    kube-state-metrics:
```
Comment (rbren) [inline stable/insights-agent/values.yaml]: is this extra key intentional?

Comment (makoscafee) [reply stable/insights-agent/values.yaml]: good catch ... those are values for `kube-state-metrics` moved them to parent `prometheus` chart

**Review by rbren — APPROVED** [review]

**Pull request merged**